### DISCUSSION TITLE
TuyaMCU Air Dehumidifier Aktobis - WDH-310EKW - Multiply the reported values?

### DISCUSSION THREAD

**Discussion by belveder79** [arendst/Tasmota#21944] [Q&A]
I successfully replaced a WR3 module from an Aktobis - WDH-310EKW Air Dehumidifier by a module based on ESP8266. I know it looks a bit hacky but it works:

![IMG_8786](https://github.com/user-attachments/assets/fd34bcbd-7bc0-4d2f-883f-518569a474a9)

![IMG_8787](https://github.com/user-attachments/assets/defaef7e-ee0b-41a9-b482-78deec7cd6ed)

Anyway, now I have configured the Temperature and Humidity dpIds with the bookmarklet, but how do I get a multiplied value?

![image](https://github.com/user-attachments/assets/70dd8c8f-4e5f-4972-afc9-2bc688a0d1db)

**Comment by belveder79** [comment]
Ok, to answer that myself:

```
HumRes 0
TempRes 0
```

solves it...

Other question: Most other settings are enums, like fan (0x00,0x01) and humidity set (0x00 to 0x06). What is the best way to map those? I saw that there is the dummy 99 id, but could this one be remapped multiple times?

**Reply by jtauscher** [reply]
Hi, did you solve these issues? Also I think dpID 101 and 102 (full water tank?) are somehow used, but dont know how to map anything.

**Comment by belveder79** [comment]
Hi, 

actually I did not solve all of the issues - it would likely need some coding inside tasmota source, which is not worth the effort (at least in my case). 

I played around by changing the value of the enums to switch the fan speed for example or to change the minimum humidity level, but this mapping resides on the control system (nodered in my case) and not in Tasmota directly. What I did though was adding a rule to publish a message to mqtt once the tank is full or when it was emptied:

```
"on TuyaReceived#Data=55AA03070005650100010176 do publish2 stat/dehumidifier/warning {\"fill\":1} endon; on TuyaReceived#Data=55AA03070005650100010075 do publish2 stat/dehumidifier/warning {\"fill\":0} endon"
```

hope that helps...
Clemens
